### PR TITLE
refractor(patient group) harmonize according 2x

### LINKED ISSUE #256: Patient Groups Module - 2.x Review
This issue tracks improvements to the patient groups client module and will take effect once #255  lands.
##### Application Structure
- [ ] Both create and update should use the same form to clean up code.
- [x] Translations still need to be implemented.
- [ ] The Patient Groups table should be designed more elegantly - possibly with cards instead of a `<table>` element.  This would be need to be presented in an issue proposal for most group management modules.
##### Form Validation
-  [ ] All forms needs to properly handle HTTP errors, displaying their translated contents in an obvious way to the client
##### CSS
- [ ] The `<table>` element does not scroll properly.  There are style hacks to make sure it has limited height which should be formalized or replaced with a different design.
##### Module Errors
- [ ] This module does not do error handling at all.  A proposal for uniform client-side error handling needs to be made and implemented before this checkbox should be ticked.
##### Binary UUIDs
- [x] The patient group module has uuids that should be binary.  It also references price lists which should contain binary uuids.
##### State Management
-  [ ] The patient groups module still uses `ng-switch` for state management.  This should be broken out into create/update/list states with ui-router to manage switching between states.
##### Code Documentation
- [ ] Use JSDoc to produce module documentation.
##### Service
- [x] Decide on a uniform way to handle patient groups in services.  Do they belong in the `PatientService` or `PatientGroupService`?  What should their API endpoint look like?
##### Design Led
- [ ]  A design for both Patient Groups and Debtor Groups should be formulated and adopted.  They will be very similar.

Once these checkboxes are all ticked, the module should be ready for `PRODUCTION READY` review.

### PR DESCRIPTION
**Pull request by jeremielodi**
Closes #3690.  Closes https://github.com/IMA-WorldHealth/bhima/issues/256.

### CODE REVIEW

File: client/src/modules/patients/groups/modal/createUpdate.html [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        <label class="label label-warning text-uppercase" translate>
+          {{ModalCtrl.action}}
+        </label>
+      </li>
+    </ol>
+  </div>
+
+  <div class="modal-body">
+    <div class="row">
+        <div class="form-group" ng-class="{ 'has-error' : patientGroupForm.$submitted && patientGroupForm.name.$invalid }">
```
Comment (mbayopanda) [inline client/src/modules/patients/groups/modal/createUpdate.html:16]: Could you wrap this `<div class="form-group"...` inside a `<div class="col-"` according the correct Bootstrap structure of rows and columns.

File: client/src/modules/patients/groups/templates/action.cell.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  </a>
+
+  <ul data-row-menu="{{ row.entity.name }}" class="dropdown-menu-right" bh-dropdown-menu-auto-dropup uib-dropdown-menu>
+    <li>
+      <a data-method="edit-record" ng-click="grid.appScope.createUpdateGroupModal(row.entity)" href>
+        <i class="fa fa-edit"></i> <span translate>FORM.BUTTONS.EDIT</span>
+      </a>
+    </li>
+    <li>
+      <a class="btn btn-xs btn-default" ui-sref="patientRegistry({
```
Comment (mbayopanda) [inline client/src/modules/patients/groups/templates/action.cell.html]: Could you remove this `class="btn btn-xs btn-default"` it is not necessary for a drop down list.

**Review by mbayopanda — COMMENTED** [review]
LGTM 👍 

I made just some little comments before to merge it

**Comment by mbayopanda** [comment]
@jeremielodi have you seen my comments ?

**Comment by mbayopanda** [comment]
@jeremielodi what is the state of this work ?

**Comment by kwilu** [comment]
Heads up @jeremielodi, we just merged some commits that conflict with the changes your made in this pull request! You can review this repository's [recent commits](https://github.com/IMA-WorldHealth/bhima/commits/master) to see where the conflicts occur. Please rebase your feature branch against the `upstream/master` branch and [resolve](https://zulip.readthedocs.io/en/latest/git/troubleshooting.html#recover-from-a-git-rebase-failure) your pull request's merge conflicts accordingly.

**Comment by mbayopanda** [comment]
@jeremielodi what is the state of this PR ?

File: client/src/modules/patients/groups/templates/action.cell.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  </a>
+
+  <ul data-row-menu="{{ row.entity.name }}" class="dropdown-menu-right" bh-dropdown-menu-auto-dropup uib-dropdown-menu>
+    <li>
+      <a data-method="edit-record" ng-click="grid.appScope.createUpdateGroupModal(row.entity)" href>
+        <i class="fa fa-edit"></i> <span translate>FORM.BUTTONS.EDIT</span>
+      </a>
+    </li>
+    <li>
+      <a data-method="view-patient" class="btn btn-xs btn-default" ui-sref="patientRegistry({
```
Comment (mbayopanda) [inline client/src/modules/patients/groups/templates/action.cell.html]: Could you remove this css class : `class="btn btn-xs btn-default"`

**Review by jniles — APPROVED** [review]
Tested it out and it works well.  Great job.

bors r+

**Pull request merged**